apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.ts

```diff
@@ -151,47 +168,44 @@ router.get(
 router.post(
   '/',
   asyncHandler(async (req, res) => {
-    const unsafe_course_instance_id = req.body.unsafe_course_instance_id;
-
     const ltiClaim = new Lti13Claim(req);
-    const authn_lti13_instance_id = req.session.authn_lti13_instance_id;
 
     // Map passed and auth lti13_instance_id through institution to course instance, or fail
-    const authorized = await queryOptionalRow(
-      sql.select_lti13_course_instance_institution,
+    const course_instance = await queryOptionalRow(
+      sql.select_lti13_institution_course_instance,
       {
-        course_instance_id: unsafe_course_instance_id,
+        course_instance_id: req.body.unsafe_course_instance_id,
         lti13_instance_id: req.params.lti13_instance_id,
-        authn_lti13_instance_id,
+        authn_lti13_instance_id: req.session.authn_lti13_instance_id,
       },
-      IdSchema,
+      CourseInstanceSchema,
     );
 
-    if (authorized == null) {
+    if (course_instance == null) {
       throw new HttpStatusError(403, 'Access denied');
     }
 
-    // Check that user is course Owner for this course instance
-    const course_instance = await selectCourseInstanceById(unsafe_course_instance_id);
-    const courses = await selectCoursesWithEditAccess({
-      user_id: res.locals.authn_user.user_id,
-      is_administrator: res.locals.authn_is_administrator,
+    const courseInstancesAllowed = await courseInstancesAllowedToLink({
+      course_id: course_instance.course_id,
+      resLocals: res.locals,
```

Comment:
Nit, for new code, it's better to pick off specific properties from `res.locals` rather than passing the entire object around.

Reply:
Addressed in 69f0ba1